orangecontrib/text/tests/test_twitter.py

```diff
@@ -0,0 +1,162 @@
+import unittest
+from unittest import mock
+import os
+
+import pickle
+from itertools import chain
+
+from orangecontrib.text import twitter
+from orangecontrib.text.corpus import Corpus
+
+
+def get_valid_credentials():
+    key = os.environ.get('TWITTER_KEY', '')
+    secret = os.environ.get('TWITTER_SECRET', '')
+    if key and secret:
+        return twitter.Credentials(key, secret)
+    return twitter.Credentials('key', 'secret')
+
+
+NO_CREDENTIALS = bool(os.environ.get('TWITTER_KEY', '')), "No credentials have been found."
+
+
+class TestCredentials(unittest.TestCase):
+
+    @unittest.skipIf(*NO_CREDENTIALS)
+    def test_valid(self):
+        credentials = get_valid_credentials()
+        self.assertTrue(credentials.valid)
+
+    def test_check_bad(self):
+        key = twitter.Credentials('bad key', 'wrong secret')
+        self.assertFalse(key.check())
+
+    def test_equal(self):
+        key1 = twitter.Credentials('key1', 'secret1')
+        key2 = twitter.Credentials('key1', 'secret1')
+        self.assertEqual(key1, key2)
+
+        key2.consumer_secret = 'key2'
+        self.assertNotEqual(key1, key2)
+
+        self.assertNotEqual(key1, None)
+
+    def test_pickle(self):
+        key1 = twitter.Credentials('key1', 'secret1')
+        pickled = pickle.dumps(key1)
+        key2 = pickle.loads(pickled)
+        self.assertEqual(key1, key2)
+
+
+# @unittest.skipIf(get_valid_credentials(), NO_CREDENTIALS)
+
+class MyCursor:
+    def __init__(self, *args, **kwargs):
+        self.statuses = pickle.load(
+            open(os.path.join(os.path.dirname(__file__), 'twitter_statuses.pkl'), 'rb'))
```

Comment:
I see that you managed to set up mocking. Congrats 👍 

One minor suggestion. Could we rather use some text format (like JSON or just txt) instead of binary pickles? They are more git-friendly and it is easier to see what's inside.